src/main/java/be/ugent/rml/Executor.java

```diff
@@ -3,26 +3,43 @@
 import be.ugent.idlab.knows.functions.agent.Agent;
 import be.ugent.rml.access.LocalFileAccess;
 import be.ugent.rml.access.RemoteFileAccess;
+import be.ugent.rml.extractor.Extractor;
 import be.ugent.rml.functions.MultipleRecordsFunctionExecutor;
+import be.ugent.rml.functions.StaticMultipleRecordsFunctionExecutor;
 import be.ugent.rml.metadata.Metadata;
 import be.ugent.rml.metadata.MetadataGenerator;
 import be.ugent.rml.records.Record;
 import be.ugent.rml.records.RecordsFactory;
 import be.ugent.rml.store.QuadStore;
 import be.ugent.rml.store.RDF4JStore;
 import be.ugent.rml.term.*;
+import org.apache.commons.collections4.MultiValuedMap;
+import org.apache.commons.collections4.multimap.AbstractListValuedMap;
+import org.apache.commons.collections4.multimap.ArrayListValuedHashMap;
+import org.apache.commons.lang3.tuple.Pair;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
 import java.io.InputStream;
 import java.util.*;
 import java.util.function.BiConsumer;
+import java.util.function.Supplier;
+import java.util.stream.IntStream;
+import java.util.stream.Stream;
 
 public class Executor {
 
     private static final Logger logger = LoggerFactory.getLogger(Executor.class);
 
+    private Map<Pair<Term, Extractor>, MultiValuedMap<List<Object>, ProvenancedTerm>> cachedRecordMap = new HashMap<>();
+
+    public static class Vocabulary {
```

Comment:
This is the RML vocabulary, it should not be here.

Reply:
Hi Dylan, is there already a class for RML vocabulary?

Reply:
IRI namespaces go into https://github.com/RMLio/rmlmapper-java/blob/master/src/main/java/be/ugent/rml/NAMESPACES.java

After that, there's no real class for that. You just need to move the `EQUAL_URI` and use string concat. If you want to make it a separate class like here, please make a separate file `RML.java` in https://github.com/RMLio/rmlmapper-java/blob/master/src/main/java/be/ugent/rml/